src/shared/components/ChannelCover/ChannelCover.tsx

```diff
@@ -0,0 +1,77 @@
+import { BackgroundPattern } from '@/components'
+import { transitions } from '@/shared/theme'
+import React, { useState } from 'react'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+import Icon from '../Icon'
+import {
+  CoverImage,
+  EditableControls,
+  EditCoverButton,
+  Media,
+  MediaWrapper,
+  RemoveCoverButton,
+} from './ChannelCover.style'
+
+type BasicChannelCoverProps = {
+  coverPhotoUrl?: string | null
+}
+type EditableProps =
+  | {
+      editable?: false
+      handleEditCover?: never
+      handleRemoveCover?: never
```

Comment:
Let's adjust the naming 

```suggestion
      onCoverEditClick?: never
      onCoverRemoveClick?: never
```